.changeset/fast-points-occur.md

```diff
@@ -0,0 +1,55 @@
+---
+'@wagmi/core': minor
+'wagmi': minor
+---
+
+**Breaking**: the shape of the `Chain` type has been modified.
+
+#### RPC URLs
+
+The `rpcUrls` shape has changed to include an array of URLs, and also the transport method (`http` or `webSocket`):
+
+```diff
+type Chain = {
+  ...
+  rpcUrls: {
+-   [key: string]: string
++   [key: string]: {
++     http: string[]
++     webSocket: string[]
++   }
+  }
+  ...
+}
+```
+
+Note that you will also need to ensure that usage is migrated:
+
+```diff
+- const rpcUrl = mainnet.rpcUrls.alchemy
++ const rpcUrl = mainnet.rpcUrls.alchemy.http[0]
+```
+
+#### Contracts
+
+The `multicall` and `ens` attributes have been moved into the `contracts` object:
+
+```diff
+type Chain = {
+  ...
+- multicall: Contract
+- ens: Contract
++ contracts: {
++   multicall3: Contract
++   ensRegistry: Contract
```

Comment:
Think it's useful to show the `Contract` type here too?

Reply:
Good idea!